gunicorn/http/wsgi.py

```diff
@@ -344,77 +347,44 @@ def write(self, arg):
         util.write(self.sock, arg, self.chunked)
 
     def can_sendfile(self):
-        return (self.cfg.sendfile and (sendfile is not None))
-
-    def sendfile_all(self, fileno, sockno, offset, nbytes):
-        # Send file in at most 1GB blocks as some operating
-        # systems can have problems with sending files in blocks
-        # over 2GB.
+        return (self.cfg.sendfile and sendfile is not None)
 
-        BLKSIZE = 0x3FFFFFFF
+    def sendfile(self, respiter):
+        if self.cfg.is_ssl or not self.can_sendfile():
+            return False
 
-        if nbytes > BLKSIZE:
-            for m in range(0, nbytes, BLKSIZE):
-                self.sendfile_all(fileno, sockno, offset, min(nbytes, BLKSIZE))
-                offset += BLKSIZE
-                nbytes -= BLKSIZE
-        else:
-            sent = 0
-            sent += sendfile(sockno, fileno, offset + sent, nbytes - sent)
-            while sent != nbytes:
-                sent += sendfile(sockno, fileno, offset + sent, nbytes - sent)
+        try:
+            fileno = respiter.filelike.fileno()
+            offset = os.lseek(fileno, 0, os.SEEK_CUR)
+            nbytes = max(os.fstat(fileno).st_size - offset, 0)
+            if self.response_length:
+                nbytes = self.response_length
+            assert(nbytes > 0)
+        except:
```

Comment:
Not sure. I limited the number of lines here very carefully. If they fail, I'm not sure it matters why they fail, the result is the same: we shouldn't use sendfile.

If you want, we can specify. It would look like `(AssertionError, OSError, io.UnsupportedOperationError)` I think.